sandbox/OpenIddict.Sandbox.Avalonia.Client/OpenIddict.Sandbox.Avalonia.Client/OpenId/AvaloniaSetup.cs

```diff
@@ -0,0 +1,161 @@
+                // address per provider, unless all the registered providers support returning an "iss"
+                // parameter containing their URL as part of authorization responses. For more information,
+                // see https://datatracker.ietf.org/doc/html/draft-ietf-oauth-security-topics#section-4.4.
+                options.UseWebProviders()
+                       .AddTwitter(options =>
+                       {
+                           options.SetClientId("bXgwc0U3N3A3YWNuaWVsdlRmRWE6MTpjaQ")
+                                  // Note: Twitter doesn't support the recommended ":/" syntax and requires using "://".
+                                  .SetRedirectUri("com.openiddict.sandbox.avalonia.client://callback/login/twitter");
+                       });
+            });
+
+        // Note: MAUI is not built on top of the .NET Generic Host and doesn't register any of
+        // the services typically found in applications using the .NET Generic Host. Since these
+        // services are required by the OpenIddict client system integration to handle callbacks
+        // and redirect protocol activations to the correct instance, custom implementations are
+        // registered here. For more information, see https://github.com/dotnet/maui/issues/2244.
+
+        services.AddSingleton<IHostEnvironment>(new HostingEnvironment
+        {
+            ApplicationName = typeof(App).Assembly.GetName().Name!
+        });
+
+        services.AddSingleton<IHostApplicationLifetime, MauiHostApplicationLifetime>();
+
+        services.AddSingleton<IMauiInitializeService>(static provider => new MauiHostedServiceAdapter(
+            ActivatorUtilities.CreateInstance<OpenIddictClientSystemIntegrationActivationHandler>(provider)));
+
+        services.AddSingleton<IMauiInitializeService>(static provider => new MauiHostedServiceAdapter(
+            ActivatorUtilities.CreateInstance<OpenIddictClientSystemIntegrationHttpListener>(provider)));
+
+        services.AddSingleton<IMauiInitializeService>(static provider => new MauiHostedServiceAdapter(
+            ActivatorUtilities.CreateInstance<OpenIddictClientSystemIntegrationPipeListener>(provider)));
+
+
+        // Register the initialization service responsible for creating the SQLite database.
+        services.AddScoped<IMauiInitializeScopedService, Worker>();
+
+        return services;
+    }
+    
+    
+    /// <summary>
+    ///     Note: Do not use the following keys in production!
+    ///     there just added so we can authenticate from Android (Maui/Avalonia), iOS (Maui/Avalonia) and Macos, since on these platforms the following does not work
+    ///     
+    ///       options.AddDevelopmentEncryptionCertificate()
+    ///         .AddDevelopmentSigningCertificate();
+    ///     
+    ///     this is because
+    ///     1. Adding an X509 certificate to the system store is not supported (throws on iOS and Android)
```

Comment:
I'm not comfortable at all using static key material (ever). Consider using `options.AddEphemeralSigningKey()` and `options.AddEphemeralEncryptionKey()` here.

In a real world app, you could use these methods (since Android and iOS apps generally live long enough in-memory for the entire flow to be completed with the app ever restarting) or you could generate RSA keys on the flight and store them securely (e.g in Keychain).